ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteImporter.java

```diff
@@ -682,16 +725,119 @@ public boolean exists(ReadOnlyDataSource dataSource) {
         }
     }
 
+    private static String getBusId(Bus bus) {
+        return bus != null ? bus.getId() : null;
+    }
+
+    private static DanglingLine getMatchingDanglingLine(DanglingLine dl1, Multimap<String, DanglingLine> danglingLinesByXnodeCode) {
+        DanglingLine dl2 = null;
+        String otherXnodeCode = dl1.getExtension(Xnode.class).getCode();
+        Iterator<DanglingLine> it = danglingLinesByXnodeCode.get(otherXnodeCode).iterator();
+        DanglingLine first = it.next();
+        if (it.hasNext()) {
+            DanglingLine second = it.next();
+            if (dl1 == first) {
+                dl2 = second;
+            } else if (dl1 == second) {
+                dl2 = first;
+            } else {
+                throw new AssertionError("Inconsistent XNODE index");
+            }
+            if (it.hasNext()) {
+                throw new UcteException("More that 2 dangling lines have the same XNODE " + dl1.getUcteXnodeCode());
+            }
+        }
+        return dl2;
+    }
+
+    private void mergeXnodeDanglingLines(Network network) {
+        Multimap<String, DanglingLine> danglingLinesByXnodeCode = HashMultimap.create();
+        for (DanglingLine dl : network.getDanglingLines()) {
+            danglingLinesByXnodeCode.put(dl.getExtension(Xnode.class).getCode(), dl);
+        }
+
+        Set<DanglingLine> danglingLinesToProcess = Sets.newHashSet(network.getDanglingLines());
+        while (!danglingLinesToProcess.isEmpty()) {
+            DanglingLine dl1 = danglingLinesToProcess.iterator().next();
+            DanglingLine dl2 = getMatchingDanglingLine(dl1, danglingLinesByXnodeCode);
+
+            if (dl2 != null) {
+                // lexical sort to always end up with same merge line id
+                String mergeLineId = dl1.getId().compareTo(dl2.getId()) < 0 ? dl1.getId() + " + " + dl2.getId()
+                                                                            : dl2.getId() + " + " + dl1.getId();
```

Comment:
Remarque: I'm not sure it's a good idea to create this kind of ID, specially if the client code expect that IDs respect the UCTE naming convention.

Reply:
Done